src/main/java/com/faforever/client/game/GenerateMapController.java

```diff
@@ -269,31 +254,41 @@ private void initCustomStyleOptions() {
   private void initSeedField() {
     seedTextField.setText(String.valueOf(generatorPrefs.getSeed()));
     generatorPrefs.seedProperty().bind(seedTextField.textProperty());
-    seedTextField.disableProperty()
-                 .bind(previousMapName.textProperty()
-                                      .isNotEmpty()
-                                      .or(generationTypeComboBox.valueProperty().isNotEqualTo(GenerationType.CASUAL))
-                                      .or(commandLineArgsText.textProperty().isNotEmpty())
-                                      .or(fixedSeedCheckBox.selectedProperty().not()));
-    seedRerollButton.disableProperty()
-                    .bind(previousMapName.textProperty()
-                                         .isNotEmpty()
-                                         .or(generationTypeComboBox.valueProperty().isNotEqualTo(GenerationType.CASUAL))
-                                         .or(commandLineArgsText.textProperty().isNotEmpty())
-                                         .or(fixedSeedCheckBox.selectedProperty().not()));
-  }
-
-  private void bindCustomStyleDisabledProperty(Node... nodes) {
+    seedTextField.disableProperty().bind(getExtendedFixedSeedNotBooleanBinding());
+    seedRerollButton.disableProperty().bind(getExtendedFixedSeedNotBooleanBinding());
+  }
+
+  private BooleanBinding getSharedBooleanBinding() {
+    return previousMapName.textProperty().isNotEmpty()
+                          .or(generationTypeComboBox.valueProperty().isNotEqualTo(GenerationType.CASUAL))
+                          .or(commandLineArgsText.textProperty().isNotEmpty());
+  }
```

Comment:
This should be renamed to like customizationAllowed or something like that

Reply:
And actually rather than having a function like this if it is going to be shared a new Property should be created and bound to this instead of using a function.

Reply:
will do, sounds much better.